http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -1331,10 +1381,37 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         }
         wsmApi = wsmClientProvider.getControlledAzureResourceApi(token)
 
-        cromwellDb <- createDatabaseInWsm(app, workspaceId, namespace, "cromwell", wsmApi, None)
-        cbasDb <- createDatabaseInWsm(app, workspaceId, namespace, "cbas", wsmApi, None)
-        tesDb <- createDatabaseInWsm(app, workspaceId, namespace, "tes", wsmApi, None)
-      } yield Some(CromwellDatabaseNames(cromwellDb, cbasDb, tesDb))
+        dbNamePrefixes = app.appType match {
+          case AppType.Cromwell          => List("cromwell", "cbas", "tes")
+          case AppType.CromwellRunnerApp => List("cromwell", "tes")
+          case _                         => List()
+        }
+
+        dbNames <-
+          dbNamePrefixes
+            .traverse(databaseNamePrefix =>
+              createDatabaseInWsm(
+                app,
+                workspaceId,
+                namespace,
+                databaseNamePrefix,
+                wsmApi,
+                None
+              )
+            )
+
+        createdDatabases = app.appType match {
+          case AppType.Cromwell =>
+            if (dbNames.size == 3) {
+              Some(CromwellDatabaseNames(dbNames.get(0).get, dbNames.get(1).get, dbNames.get(2).get))
+            } else None
+          case AppType.CromwellRunnerApp =>
+            if (dbNames.size == 2) {
+              Some(CromwellRunnerDatabaseNames(dbNames.get(0).get, dbNames.get(1).get))
+            } else None
```

Comment:
@cjllanwarne and I discussed the length checking in #3603 and decided against it to have a louder failure if the length is wrong rather than returning None which would progress through further and fail more quietly

Reply:
Ah I see. So I should replace `dbNames.get(1).get` with `dbNames.apply(1)` right?

Reply:
I think we can get rid of the if dbNames.size checks

Reply:
Yes I have removed that check. I was curious about `.apply(N)` because `.get(N).get` looks ugly